GitVersionCore.Tests/Helpers/GitTestExtensions.cs

```diff
@@ -39,6 +37,48 @@ public static Commit[] MakeCommits(this IRepository repository, int numCommitsTo
             .ToArray();
     }
 
+    public static Commit CreateFileAndCommit(this IRepository repository, string relativeFileName, DateTimeOffset dateTimeOffset = default(DateTimeOffset))
+    {
+        if (dateTimeOffset == default(DateTimeOffset))
+        {
+            dateTimeOffset = DateTimeOffset.Now;
+        }
+
+        var randomFile = Path.Combine(repository.Info.WorkingDirectory, relativeFileName);
+        if (File.Exists(randomFile))
+        {
+            File.Delete(randomFile);
+        }
+
+        File.WriteAllText(randomFile, Guid.NewGuid().ToString());
+
+        // GHK: 2015-01-18: I know it's very ugly, but somehow we need to retry here otherwise "there is nothing to commit"
+        var retryCount = 3;
```

Comment:
@SimonCropp @JakeGinnivan `EmptyCommitException` is raised when the tree about to be committed matches the tree from the parent commit. This means that no modification has been detected. My guess is that is related to a known issue in libgit2 (see https://github.com/libgit2/libgit2sharp/pull/688).

Meanwhile, in order to work around this, just make sure that your files are of random length